File: app/src/main/java/com/bq/zowi/controllers/BTConnectionController.java
package com.bq.zowi.controllers;

import android.bluetooth.BluetoothDevice;
import java.io.InputStream;
import java.io.OutputStream;
import rx.Observable;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface BTConnectionController {
    Single<Void> connect(BluetoothDevice bluetoothDevice, boolean z);

    void enabledReadingIncomingData();

    InputStream getBTInputStream();

    OutputStream getBTOutputStream();

    Observable<Integer> getConnectionStatusObservable();

    Observable<String> getReceivedMessageObservable();

    boolean isConnected();

    boolean isSendingHexToZowi();

    Observable<Integer> sendHexToZowi(InputStream inputStream);

    void sendMessage(String str);

    void stopConnection();
}

File: zowi-core/src/main/kotlin/com/bq/zowi/api/BTConnectionController.kt
package com.bq.zowi.api

import io.reactivex.Observable
import io.reactivex.Single
import java.io.InputStream
import java.io.OutputStream

interface BTConnectionController {
    fun connect(device: DeviceHandle, duplex: Boolean): Single<Void>
    fun enabledReadingIncomingData()
    fun getBTInputStream(): InputStream
    fun getBTOutputStream(): OutputStream
    fun getConnectionStatusObservable(): Observable<Int>
    fun getReceivedMessageObservable(): Observable<String>
    fun isConnected(): Boolean
    fun isSendingHexToZowi(): Boolean
    fun sendHexToZowi(inputStream: InputStream): Observable<Int>
    fun sendMessage(message: String)
    fun stopConnection()
}
